### PR TITLE
Add Backstage metadata

### PR DESCRIPTION
**Pull request by vzctl**
@sept-en Could you please review the precompiles piece and add description?
@joshuajbouw please check and feel free to add more info!

### CODE REVIEW

**Review by joshuajbouw — APPROVED** [review]

**Comment by joshuajbouw** [comment]
Does anything more need to be added to this @vzctl ? I'm not too familiar with what could be added.

File: engine/.catalog-info.yaml [hunk 1]
```diff
@@ -0,0 +1,20 @@
+---
+apiVersion: backstage.io/v1alpha1
+kind: Component
+metadata:
+  name: aurora-engine
+  title: Aurora (engine)
+  tags:
```
Comment (joshuajbouw) [inline engine/.catalog-info.yaml:7]:
```suggestion
  description: |-
    The Aurora engine
  tags:
```

File: engine-precompiles/.catalog-info.yaml [hunk 2]
```diff
@@ -0,0 +1,37 @@
+---
+apiVersion: backstage.io/v1alpha1
+kind: Component
+metadata:
+  name: aurora-engine-precompiles
+  tags:
```
Comment (joshuajbouw) [inline engine-precompiles/.catalog-info.yaml:6]:
```suggestion
  description: |-
    EVM precompiles for the Aurora engine
  tags:
```

**Review by joshuajbouw — CHANGES_REQUESTED** [review]

**Comment by joshuajbouw** [comment]
The `engine-standalone-storage` and `engine-standalone-tracing` currently is owned by the Protocol team. I figure that will change in the future, however.

`engine-sdk` is owned by us, and serves as our SDK for interfacing with NEAR. It is a component.

`engine-transactions` is owned by us and includes all the different kinds of ETH transactions.

`engine-types` is a shared types library that all other modules use.

**Comment by vzctl** [comment]
All looks good to me! @sept-en is it good to go from your side?

File: etc/eth-contracts/.catalog-info.yaml [hunk 3]
```diff
@@ -0,0 +1,21 @@
+---
+apiVersion: backstage.io/v1alpha1
+kind: Component
+metadata:
+  name: aurora-engine-token
+  description: |-
+    Aurora token deployed on the Aurora network
```
[the remark this answers is not in the document]
Comment (sept-en) [reply etc/eth-contracts/.catalog-info.yaml:7]: I'd say rather

```suggestion
    Aurora ERC-20 contract for bridged tokens
```

File: etc/eth-contracts/.catalog-info.yaml [hunk 4]
```diff
@@ -0,0 +1,21 @@
...
+  links: []
+  annotations:
+    aurora.dev/security-tier: "1"
+spec:
+  owner: engine-team
+  type: contract
+  lifecycle: production
+  system: aurora-engine
+  deployedAt:
+    - contract:aurora/mainnet/0x8bec47865ade3b172a928df8f990bc7f2a3b9f79
```
Comment (sept-en) [inline etc/eth-contracts/.catalog-info.yaml:20]: This lists only one particular contract for $AURORA token. Meanwhile, there are plenty of such tokens deployed. How should we deal with this case?

**Review by sept-en — CHANGES_REQUESTED** [review]

File: etc/eth-contracts/.catalog-info.yaml (hunk 4 above)
Comment (vzctl) [reply etc/eth-contracts/.catalog-info.yaml:20]: It's a very good point and we will need to implement some discovery mechanism to ingest all deployed contracts automatically from the chain.

Let's keep this out of scope for now - a bit too complicated to solve in the short term.

**Review by sept-en — APPROVED** [review]

**Review by joshuajbouw — APPROVED** [review]

**Pull request merged**